src/main/java/org/jabref/gui/desktop/os/Linux.java

```diff
@@ -53,17 +56,18 @@ public void openFileWithApplication(String filePath, String application) throws
         } else {
             openWith = new String[] {"xdg-open"};
         }
-
         String[] cmdArray = new String[openWith.length + 1];
         System.arraycopy(openWith, 0, cmdArray, 0, openWith.length);
         cmdArray[cmdArray.length - 1] = filePath;
-        Process p = Runtime.getRuntime().exec(cmdArray);
+
+        Process process = Runtime.getRuntime().exec(cmdArray);
```

Comment:
I guess the whole process starting has to be done in a different thread...

Reply:
That's totally weird as to what I read about is that the process is executed and only if you call waitFor the app blocks. https://www.javaworld.com/article/2071275/when-runtime-exec---won-t.html
And the stream gobbler is there to capture the output

Reply:
Strange... can you please try the ProcessBuilder, which anyway seems to be the preferred way to start subprocesses. https://docs.oracle.com/javase/6/docs/api/java/lang/ProcessBuilder.html